### PR TITLE
Update cache check to use django.core.cache.caches dictionary

### PR DESCRIPTION
**Pull request by justinsacbibit**
@mwarkentin I did a `pip install -e` on accounting and it seems to work well. I don't have a project with Django >= 1.7 and a cache to test with

Fixes https://github.com/mwarkentin/django-watchman/issues/47

### CODE REVIEW

File: watchman/checks.py [hunk 1]
```diff
@@ -22,7 +23,13 @@ def _check_cache(cache_name):
     key = 'django-watchman-{}'.format(uuid.uuid4())
     value = 'django-watchman-{}'.format(uuid.uuid4())
 
-    cache = get_cache(cache_name)
+    # As of Django 1.7, django.core.cache.caches replaces
+    # django.core.cache.get_cache
+    if django.VERSION < (1, 7):
+        cache = django_cache.get_cache(cache_name)
+    else:
+        cache = django_cache.caches[cache_name]
```
Comment (mwarkentin) [inline watchman/checks.py]: @justinsacbibit Thanks for this! 

Thinking about this a bit more, I think we should actually move this into a small function in `watchman/utils.py` to keep this function a bit cleaner.

Thinking something like:

```python
# checks.py

from watchman.utils import get_cache

cache = get_cache(cache_name)

# watchman/utils.py
def get_cache:
    if django.version < (1, 7):
        return django_cache.get_cache(cache_name)
    else:
        return django_cache.caches[cache_name]
```

**Comment by justinsacbibit** [comment]
@mwarkentin Refactored the version checking logic into a util method. This is yet to be manually tested on Django 1.7+

File: tests/test_views.py [hunk 2]
```diff
@@ -72,11 +72,18 @@ def test_check_database_handles_exception(self):
...
+
         response = checks._check_cache('foo')
         self.assertFalse(response['foo']['ok'])
         self.assertIn(response['foo']['error'], expected_error)
 
+    @patch('watchman.utils.get_cache')
+    def test_check_cache_uses_util_get_cache(self, get_cache_mock):
+        checks._check_cache('foo')
+
+        get_cache_mock.assert_called_once_with('foo')
```
Comment (mwarkentin) [inline tests/test_views.py]: @justinsacbibit I'm not sure we need this test.

File: tests/test_utils.py [hunk 3]
```diff
@@ -24,6 +28,15 @@ def assertListsEqual(self, list1, list2):
...
             self.assertItemsEqual(list1, list2)
 
+    @patch('django.core.cache.get_cache')
+    @patch('watchman.utils.django')
+    def test_get_cache_less_than_django_17(self, django_mock, get_cache_mock):
+        django_mock.VERSION = (1, 6, 6, 'final', 0)
+
+        get_cache('foo')
+
+        get_cache_mock.assert_called_once_with('foo')
```
Comment (mwarkentin) [inline tests/test_utils.py]: @justinsacbibit Any chance you could add a mirror test for django >= 1.7?

**Pull request merged**